reframe/core/environments.py

```diff
@@ -79,8 +79,9 @@ def load(self):
             if rt.modules_system.is_module_loaded(m):
                 self._preloaded.add(m)
 
-            self._conflicted += rt.modules_system.load_module(m, force=True)
-            for conflict in self._conflicted:
+            module_conflicts = rt.modules_system.load_module(m, force=True)
+            self._conflicted += module_conflicts
```

Comment:
Move this assignment after the loop. Placed here, it kind of lets you expect that `self._conflicted` will be used just right after.